pkg/flist/flist.go

```diff
@@ -160,6 +160,11 @@ func (f *flistModule) mount(name, url, storage string, opts pkg.MountOptions) (s
 	sublog := log.With().Str("url", url).Str("storage", storage).Logger()
 	sublog.Info().Msg("request to mount flist")
 
+	if len(opts.Type) == 0 {
+		// sanity check in case type is not set always use hdd
+		opts.Type = pkg.HDDDevice
```

Comment:
I would return an error instead. At this point we should have the information, not having it is mostly sign of a bad request from the user.

Reply:
actually this is an internal call. the user does not have a direct access to flist.Mount.

Reply:
But in that case, since this is optional (for read/write flists) I should move this check when it's needed.